### PR TITLE
Ensure dataplane is deployed successfully

### PR DESCRIPTION
**Pull request by fao89**

### CODE REVIEW

**Comment by fao89** [comment]
getting a weird error on dataplane logs:
```
Error loading key "/runner/artifacts/dataplane-deployment-configure-network-edpmadoption/ssh_key_data": error in libcrypto
```
https://review.rdoproject.org/zuul/build/dded5efd57c44c31be69774812293427/log/controller/pod/dataplane-deployment-configure-network-edpmadoption-zv78k-logs.txt

**Comment by fao89** [comment]
maybe the key is lacking a new line
https://github.com/openssl/openssl/issues/13443#issuecomment-1354750805

**Comment by fao89** [comment]
recheck

**Comment by jistr** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
> Build failed (check pipeline). For information on how to proceed, see http://docs.openstack.org/infra/manual/developers.html#automated-testing
> 
> https://review.rdoproject.org/zuul/buildset/6fd7dd4f747c41cbaf3551440f19bd58
> 
> x [centos-9-standalone-to-crc-data-plane-adoption ](https://review.rdoproject.org/zuul/build/61bfa35f0c5343b19dfff26a3924a3d9) **POST_FAILURE** in 2h 14m 10s heavy_check_mark [data-plane-adoption-github-rdo-centos-9-crc-single-node ](https://review.rdoproject.org/zuul/build/1a34e62798144e2083847c394e28d488) SUCCESS in 1h 42m 16s

@marios I managed to get the single node working, but the other job keeps failing (need to investigate more)

**Comment by fao89** [comment]
recheck

File: tests/roles/dataplane_adoption/tasks/main.yaml [hunk 1]
```diff
@@ -1,3 +1,73 @@
...
+        dnsDomain: ctlplane.example.com
+        subnets:
+        - name: subnet1
+          allocationRanges:
+          - end: 192.168.122.120
+            start: 192.168.122.100
+          - end: 192.168.122.200
+            start: 192.168.122.150
+          cidr: 192.168.122.0/24
+          gateway: 192.168.122.1
```
Comment (fao89) [inline tests/roles/dataplane_adoption/tasks/main.yaml:23]:
```
{"level":"info","ts":"2023-07-28T15:32:59Z","msg":"Reconciling IPSet","controller":"openstackdataplanerole","controllerGroup":"dataplane.openstack.org","controllerKind":"OpenStackDataPlaneRole","OpenStackDataPlaneRole":{"name":"edpmadoption","namespace":"openstack"},"namespace":"openstack","name":"edpmadoption","reconcileID":"edfe14bb-90ca-46eb-ad13-e895d1d4178f","ObjectType":"*v1beta1.OpenStackDataPlaneRole","ObjectNamespace":"openstack","ObjectName":"edpmadoption"}
{"level":"error","ts":"2023-07-28T15:32:59Z","msg":"Reconciler error","controller":"openstackdataplanerole","controllerGroup":"dataplane.openstack.org","controllerKind":"OpenStackDataPlaneRole","OpenStackDataPlaneRole":{"name":"edpmadoption","namespace":"openstack"},"namespace":"openstack","name":"edpmadoption","reconcileID":"edfe14bb-90ca-46eb-ad13-e895d1d4178f","error":"admission webhook \"vipset.kb.io\" denied the request: IPSet.network.openstack.org \"standalone\" is invalid: spec.networks[0].fixedIP: Invalid value: \"38.102.83.129\": address not in IP address prefix (CIDR) 192.168.122.0/24","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235"}
```
that's why data-plane-adoption-github-rdo-centos-9-crc-single-node  works and centos-9-standalone-to-crc-data-plane-adoption breaks

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

File: tests/roles/dataplane_adoption/tasks/main.yaml [hunk 2]
```diff
@@ -205,14 +231,13 @@
...
+              edpm_selinux_mode: enforcing
+              plan: overcloud
     EOF
+
+- name: wait for dataplane to be ready
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc wait --for condition=Ready osdp/openstack --timeout=30m
+  when: "edpm_node_ip.startswith('192.168.122')"
```
Comment (fao89) [inline tests/roles/dataplane_adoption/tasks/main.yaml:243]: temporary workaround for `centos-9-standalone-to-crc-data-plane-adoption`

Comment (marios) [reply tests/roles/dataplane_adoption/tasks/main.yaml:243]: do you want to add a #TODO(fao89): 
so we can revisit that then?
but what will the proper fix be? why is it temporary? i think we will only ever run that for the single node job?

File: tests/roles/dataplane_adoption/tasks/main.yaml [hunk 3]
```diff
@@ -183,8 +210,7 @@
               edpm_ovn_metadata_agent_metadata_agent_DEFAULT_metadata_proxy_shared_secret: 1234567842
               edpm_ovn_metadata_agent_DEFAULT_bind_host: 127.0.0.1
               edpm_chrony_ntp_servers:
-              - clock.redhat.com
-              - clock2.redhat.com
+              - pool.ntp.org
```
Comment (marios) [inline tests/roles/dataplane_adoption/tasks/main.yaml:213]: we'll need to make this configurable

for downstream jobs it will indeed be clock.redhat.com

**Review by marios — COMMENTED** [review]
nice one getting both jobs green \o/

File: tests/roles/dataplane_adoption/tasks/main.yaml (hunk 2 above)
Comment (fao89) [reply tests/roles/dataplane_adoption/tasks/main.yaml:243]: yes, I will add a TODO
I'm learning about the [networking config](https://github.com/openstack-k8s-operators/docs/blob/main/networking.md) and I noticed that with the current install_yamls way, many places expect the ip range to be under `192.168.122`, which would require changes in other test roles and on CI config. As we have a job already working, I think we can enable it, and I can work on the other job in a follow-up PR

File: tests/roles/dataplane_adoption/tasks/main.yaml (hunk 3 above)
Comment (fao89) [reply tests/roles/dataplane_adoption/tasks/main.yaml:213]: I'll add a variable for it

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
it is getting stuck on `make standalone` 😢

**Review by marios — APPROVED** [review]
lgtm and tests passed on the latest run https://review.rdoproject.org/zuul/buildset/568f2d38f65d4188a19079ccfdf261a2

**Pull request merged**